docs/architecture/adr-043-nft-module.md

```diff
@@ -0,0 +1,150 @@
+# ADR 43: NFT Module
+
+## Changelog
+
+- 05.05.2021: Initial Draft
+
+## Status
+
+DRAFT
+
+## Abstract
+
+This ADR defines a generic NFT module of `x/nft` which supports NFTs as a `proto.Any` and contains `NFT` as the default implementation.
+
+## Context
+``
+NFTs are more digital assets than only crypto arts, which is very helpful for accruing value to ATOM. As a result, Cosmos Hub should implement NFT functions and enable a unified mechanism for storing and sending the ownership representative of NFTs ss discussed in https://github.com/cosmos/cosmos-sdk/discussions/9065.
+
+As was discussed in [#9065](https://github.com/cosmos/cosmos-sdk/discussions/9065), several potential solutions can be considered:
+- irismod/nft and modules/incubator/nft
+- CW721
+- DID NFTs
+- interNFT
+
+Since NFTs functions/use cases are tightly connected with their logic, it is almost impossible to support all the NFTs' use cases in one Cosmos SDK module by defining and implementing different transaction types.
+
+Considering generic usage and compatibility of interchain protocols including IBC and Gravity Bridge, it is preferred to have a very simple NFT module design which only stores NFTs by id and owner. 
+
+This design idea can enable composability that application-specific functions should be managed by other modules on Cosmos Hub or on other Zones by importing the NFT module.
+
+For example, NFTs can be grouped into collections in a collectibles module to define the behaviors such as minting, burning, transferring, etc.
+
+The current design is based on the work done by [IRISnet team](https://github.com/irisnet/irismod/tree/master/modules/nft) and an older implementation in the [Cosmos repository](https://github.com/cosmos/modules/tree/master/incubator/nft).
+
+
+## Decision
+
+We will create a module `x/nft` which only stores NFTs by id and owner.
+
+The interface for the `x/nft` module:
+
+```go
+// NFT is an interface used to store NFTs at a given id.
+type NFT interface {
+    GetId() string // define a simple identifier for the NFT 
+    GetOwner() sdk.AccAddress // the current owner's address
+    GetData() string // specialized metadata to accompany the nft
```

Comment:
If this references metadata, I'd call it that.

Reply:
My thinking is that `Data` can include metadata, such as provenance, that'd be useful for some service down the line. As well, the field could contain actual nft data (e.g., hash or seed for generators). It'd be up to each NFT module to specify and interpret its contents. For this `x/nft` module, the value in this field doesn't provide additional context required to manage nft records. Although this means that technically the field can be removed, I think there's value in specifying the field to provide a place for basic informational/UI purposes.

Reply:
I would make note of that then because this ADR only seems to address metadata as the primary use case.